### ISSUE TITLE
Declaration line in Lua signature help looks entirely like an @error

### ISSUE THREAD

**Issue by przepompownia** [gbprod/nord.nvim#48]
Hey @gbprod 

please look at some signature help window with content from lua-ls:

currently I have no idea how to avoid displaying the whole declaration line as an error:
![image](https://github.com/gbprod/nord.nvim/assets/11404453/78d2e4f9-638f-4f18-a54b-ca244f45fb3b)

For some reason, another colorscheme displays the same in a better way:
![image](https://github.com/gbprod/nord.nvim/assets/11404453/5135f8cc-de62-43c0-9456-4742967c57b0)

Can you reproduce this look?

**Comment by gbprod** [comment]
Thanks for your feedback. It doesn't seem to comes from `nord.nvim` in particular but maybe more from tree-sitter and the markdown tree-sitter. I think that the colorscheme you used to compare doesn't support tree-sitter `@error` highlight.
If you try with another theme supporting tree-sitter, you have the same behavior, eg. with tokyonight :

![2023-10-17-095018_984x221_scrot](https://github.com/gbprod/nord.nvim/assets/3751019/f8f23fb4-c5b1-48cf-a296-4a572515eeb2)

I'm not sure of where we should post an issue for that :sweat_smile: and I don't actually have any solution for that :/

**Comment by przepompownia** [comment]
Also thank you for looking into the issue. The light colorscheme shown above is https://github.com/navarasu/onedark.nvim and supports `@error` highlight. With
```lua
require('onedark').setup {
  style = 'light',
  highlights = {
    ['@error'] = {fg = '#884400'},
  }
}
```
`@error` highlight spans only `function` keyword there:
![image](https://github.com/gbprod/nord.nvim/assets/11404453/e71c3a79-98ea-4bac-8dc9-0d7a14e03132)

After similar change
```lua
require('nord').setup {
  diff = {mode = 'fg'},
  on_highlights = function (highlights, _colors)
    highlights['@error'] = {fg = '#884400'}
  end,
}
```
I got
![image](https://github.com/gbprod/nord.nvim/assets/11404453/4a0f1db2-5728-47c1-a81e-7e3ecd3a8e4c)
but still cannot explain why it behaves in this way.

**Comment by gbprod** [comment]
I can explain what happens :

The buffer with the hover content is considered as `markdown`, so the `markdown` tree-sitter parser is applied. 
With `lua_ls`, the lsp hover contains a lua block code inside. Eg : 
```
```lua
function vim.tbl_extend(behavior: string, ...table)
  -> Merged: table
```

This part of the markdown is considered as lua so the lua tree-sitter is used for this part (it's [language injection](https://tree-sitter.github.io/tree-sitter/syntax-highlighting#language-injection)) but the lua code is not correct so, tree-sitter parse it as an error and apply the `@error` highlight group.

I think it's possible to avoid it using a specific tree-sitter rule. I will try to find someting.

Using `onedark`, the behavior is the same but, as the highlight group only apply a foreground color, it's less annoying.

I hope my explanation is clear :sweat_smile:

**Comment by przepompownia** [comment]
Yes, it's clear for me. I looked earlier on `:InspectTree` (built-in playground, after `<C-w>T` on the float window) that `@error` covers the whole line.

Indeed, after customizing `bg` on both colorschemes I see background covering the whole line. 
![image](https://github.com/gbprod/nord.nvim/assets/11404453/e005280a-6425-438c-94e1-d219e0d07624)

The first parameter is considered as active so it's highlighted separately (https://github.com/neovim/neovim/commit/2eecb1b85dbb0e02ec40e95d28d3cc4720040f01) but the rest isn't red (only background setting works). 

It's possible that the rest is **highlighted later** and **overrides `@error`**. The later highlights usually **have no `bg`** thus the setting from colorscheme survives if existed before.

To test it within float window I tried to `call nvim_buf_add_highlight(0, -1, 'DiffDelete', 1, 26, 28)` and `call nvim_buf_add_highlight(0, -1, '@text.uri', 1, 21, 24) ` (only the first has `bg`). 

We probably have an explanation of what's going on so I can close this issue. Thanks for looking into the issue.
